Mathlib/Data/Real/EReal.lean

```diff
@@ -445,6 +445,62 @@ theorem eq_bot_iff_forall_lt (x : EReal) : x = ⊥ ↔ ∀ y : ℝ, x < (y : ERe
     exact ⟨x.toReal, coe_toReal_le h⟩
 #align ereal.eq_bot_iff_forall_lt EReal.eq_bot_iff_forall_lt
 
+/-! ### Intervals and coercion from reals -/
+
+lemma exists_between_ofReal_left {x : EReal} {z : ℝ} (h : x < z) : ∃ y : ℝ, x < y ∧ y < z := by
+  obtain ⟨a, ha₁, ha₂⟩ := exists_between h
+  induction' a using EReal.rec with a₀
+  · simp only [not_lt_bot] at ha₁
+  · exact ⟨a₀, by exact_mod_cast ha₁, by exact_mod_cast ha₂⟩
+  · simp only [not_top_lt] at ha₂
+
+lemma exists_between_ofReal_right {x : ℝ} {z : EReal} (h : x < z) : ∃ y : ℝ, x < y ∧ y < z := by
+  obtain ⟨a, ha₁, ha₂⟩ := exists_between h
+  induction' a using EReal.rec with a₀
+  · simp only [not_lt_bot] at ha₁
+  · exact ⟨a₀, by exact_mod_cast ha₁, by exact_mod_cast ha₂⟩
+  · simp only [not_top_lt] at ha₂
+
+lemma Icc_of_Real {x y : ℝ} : Real.toEReal '' Set.Icc x y = Set.Icc ↑x ↑y := by
```

Comment:
These should be `@[simp]`. BTW, do we have `Set.preimage` lemmas? If no, then please add them too.

Reply:
```suggestion
lemma image_toEReal_Icc {x y : ℝ} : Real.toEReal '' Set.Icc x y = Set.Icc ↑x ↑y := by
```

Reply:
You mean like `Real.toEReal ⁻¹' Set.Ixy ... = ...` ? I don't think so.

Reply:
```suggestion
lemma image_coe_Icc {x y : ℝ} : Real.toEReal '' Set.Icc x y = Set.Icc ↑x ↑y := by
```
is the name I would guess for this lemma

Reply:
OK; I'll change the names.
Regarding preimage lemmas, I think there will be 24 cases (for I{c|o}{c|o} one has to distinguish cases according to whether the left end is bot or not/the right end is top or not --> 4*4 cases, and for I{c|o}i whether the left end is bot or not --> 2*2 cases, similarly for Ii{c|o} --> 2*2 cases). I'll add them unless somebody objects.

Reply:
I think they all make sense to have.